fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HiveMetadataOps.java

```diff
@@ -0,0 +1,188 @@
+import org.apache.doris.datasource.ExternalMetaIdMgr;
+import org.apache.doris.datasource.jdbc.client.JdbcClient;
+import org.apache.doris.datasource.jdbc.client.JdbcClientConfig;
+
+import com.google.common.base.Preconditions;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.api.FieldSchema;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class HiveMetadataOps implements ExternalMetadataOps {
+    private static final Logger LOG = LogManager.getLogger(HiveMetadataOps.class);
+    private static final int MIN_CLIENT_POOL_SIZE = 8;
+    private JdbcClientConfig jdbcClientConfig;
+    private HiveConf hiveConf;
+    private HMSExternalCatalog catalog;
+    private HMSCachedClient client;
+
+    public HiveMetadataOps(HiveConf hiveConf, JdbcClientConfig jdbcClientConfig, HMSExternalCatalog catalog) {
+        this.catalog = catalog;
+        this.hiveConf = hiveConf;
+        this.jdbcClientConfig = jdbcClientConfig;
+        this.client = createCachedClient(hiveConf,
+                Math.max(MIN_CLIENT_POOL_SIZE, Config.max_external_cache_loader_thread_pool_size), jdbcClientConfig);
+    }
+
+    public HMSCachedClient getClient() {
+        return client;
+    }
+
+    public static HMSCachedClient createCachedClient(HiveConf hiveConf, int thriftClientPoolSize,
+                                                     JdbcClientConfig jdbcClientConfig) {
+        if (hiveConf != null) {
+            return new ThriftHMSCachedClient(hiveConf, thriftClientPoolSize);
+        }
+        Preconditions.checkNotNull(jdbcClientConfig, "hiveConf and jdbcClientConfig are both null");
+        String dbType = JdbcClient.parseDbType(jdbcClientConfig.getJdbcUrl());
+        switch (dbType) {
+            case JdbcResource.POSTGRESQL:
+                return new PostgreSQLJdbcHMSCachedClient(jdbcClientConfig);
+            default:
+                throw new IllegalArgumentException("Unsupported DB type: " + dbType);
+        }
+    }
+
+    @Override
+    public void createDb(CreateDbStmt stmt) throws DdlException {
+        String fullDbName = stmt.getFullDbName();
+        Map<String, String> properties = stmt.getProperties();
+        long dbId = Env.getCurrentEnv().getNextId();
+        try {
+            HiveDatabaseMetadata catalogDatabase = new HiveDatabaseMetadata();
+            catalogDatabase.setDbName(fullDbName);
+            catalogDatabase.setProperties(properties);
+            if (properties.containsKey("location_uri")) {
+                catalogDatabase.setLocationUri(properties.get("location_uri"));
+            }
+            catalogDatabase.setComment(properties.getOrDefault("comment", ""));
+            client.createDatabase(catalogDatabase);
+            catalog.onRefresh(true);
+            catalog.registerDatabase(dbId, fullDbName);
```

Comment:
Is it necessary to call `registerDatabase()` after `onRefresh()`?
because after refresh, if we want to get db, it will automatically sync all dbs.